monix-reactive/shared/src/main/scala/monix/reactive/Observable.scala

```diff
@@ -6056,6 +6056,19 @@ object Observable extends ObservableDeprecatedBuilders {
     }
   }
 
+  /** Given an observable sequence and associated priorities, it combines them
+    * into a new observable, preferring higher-priority sources when multiple
+    * sources have items available. If items are available from sources with the
+    * same priority, the order is undefined.
+    */
+  def mergePrioritizedList[A](sources: Observable[A]*)(priorities: Seq[Int]): Observable[A] = {
```

Comment:
Very interesting implementation @ctoomey.

I do have one suggestion about this signature ... it's not clear that the `priorities` should match the sources and there's no reason for it.

How about ...

```scala
def mergePrioritizedList[A](sources: (Int, Observable[A])*): Observable[A]
```

Then you could do ...

```scala
Observable.mergePrioritizedList(
  1 -> observable1,
  2 -> observable2,
  //...
)
```

That way the signature is clearer, and you don't need that runtime `assert` in the implementation.

Reply:
Ya I'd prefer that signature too and in fact had it that way originally, but decided to match the way Akka streams did it. I'll change it.